dapp_manager/cli.py

```diff
@@ -0,0 +1,190 @@
+
+    return wrapped
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option(
+    "--config",
+    "-c",
+    required=True,
+    type=Path,
+    help="Path to the file containing yagna-specific config.",
+)
+@click.argument(
+    "descriptors",
+    nargs=-1,
+    required=True,
+    type=Path,
+)
+@_capture_api_exceptions
+def start(descriptors: Tuple[Path], *, config: Path):
+    """Start a new app using the provided descriptor and config files."""
+    dapp = DappManager.start(*descriptors, config=config)
+    print(dapp.app_id)
+
+
+@cli.command()
+@_capture_api_exceptions
+def list():
+    """List known app IDs (both active and dead).
+
+    The results are sorted by the apps' creation time.
+    """
+    app_ids = DappManager.list()
+    if app_ids:
+        print("\n".join(app_ids))
+
+
+@cli.command()
+@_capture_api_exceptions
+def prune():
+    """Remove data for non-running apps.
+
+    This removes all data related to those apps, including logs, state etc.
+    """
+    app_ids = DappManager.prune()
+    if app_ids:
+        print("\n".join(app_ids))
+
+
+@cli.command()
+@click.option(
+    "--timeout",
+    "-t",
+    type=int,
+    default=10,
+    help="Specify a shutdown timeout in seconds. Successful shutdown is indicated by"
+    " the app_id print",
+)
+@_with_app_id
+@_capture_api_exceptions
+def stop(*, app_id: str, timeout: int):
+    """Stop the given app gracefully.
+
+    Requests a stop from the given app through a SIGINT.
+    Optionally, a timeout (in seconds) may be given with --timeout flag.
+    """
+    dapp = DappManager(app_id)
+    if dapp.stop(timeout):
+        print(app_id)
+
+
+@cli.command()
+@_with_app_id
+@_capture_api_exceptions
+def kill(*, app_id):
+    """Stop the given app forcibly.
+
+    Stops the app's process using SIGKILL.
+    """
+    dapp = DappManager(app_id)
+    dapp.kill()
+    print(app_id)
+
+
+@cli.command()
+@_with_app_id
+@click.argument("service", required=True, type=str)
+@click.argument(
+    "command",
+    nargs=-1,
+    required=True,
+    type=str,
+)
+@click.option(
+    "--timeout",
+    type=int,
+    default=60,
+)
+@_capture_api_exceptions
+def exec(*, app_id, service, command, timeout):
+    dapp = DappManager(app_id)
+    dapp.exec_command(service, command, timeout)
+
+
+@cli.command()
+@_with_app_id
+@click.argument("file-type", type=click.Choice(["state", "data", "log", "stdout", "stderr"]))
+@click.option(
+    "--ensure-alive/--no-ensure-alive",
+    default=True,
+)
+@click.option(
+    "-f",
+    "--follow",
+    is_flag=True,
+    default=False,
+)
+@_capture_api_exceptions
+def read(app_id: str, file_type: RunnerReadFileType, ensure_alive: bool, follow: bool):
+    """Read output from the given app."""
+
+    dapp = DappManager(app_id)
+
+    read_function: Callable = dapp.read_file_follow if follow else dapp.read_file
+
+    for chunk in read_function(file_type, ensure_alive=ensure_alive):
+        print(chunk, end="")
```

Comment:
I am not a fan of new `read_file` interface. I would keep the `read_file` as a return function. `sample_usage.py` showcases that it got more complicated and I cannot find a reason why.

Here we could simply implement it like this 
```python
if follow is True:
    for chunk in read_function(file_type, ensure_alive=ensure_alive):
        print(chunk, end="")
else:
    print(dapp.read_file)
```